packages/deploy/src/validator.ts

```diff
@@ -13,17 +13,19 @@ export function parseAndValidate(input: string): {
   doc: ProjectSpec;
 } {
   let errors: Error[] = [];
-  let keys: string[] = [];
+  let workflowKeys: Set<string> = new Set();
```

Comment:
It is strange to have a set here. Normally a set would ensure that a value is only added once.

But here, we have to check for membership of the set before we add it anyway. It's no different to using an array - and an array is a simpler and more commonly used structure.

So I'd prefer both of these key sets to actually be arrays - it's just simpler.